docs/samples/Microsoft.ML.Samples/Dynamic/IidSpikeDetectorTransform.cs

```diff
@@ -83,5 +87,84 @@ public static void IidSpikeDetectorTransform()
             // 0       5.00    0.50
             // 0       5.00    0.50
         }
+
+        public static void IidSpikeDetectorPrediction()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var ml = new MLContext();
+
+            // Generate sample series data with a spike
+            const int size = 10;
+            var data = new List<IidSpikeData>(size);
+            for (int i = 0; i < size / 2; i++)
+                data.Add(new IidSpikeData(5));
+            // This is a spike
+            data.Add(new IidSpikeData(10));
+            for (int i = 0; i < size / 2; i++)
+                data.Add(new IidSpikeData(5));
+
+            // Convert data to IDataView.
+            var dataView = ml.CreateStreamingDataView(data);
+
+            // Setup IidSpikeDetector arguments
+            string outputColumnName = "Prediction";
+            string inputColumnName = "Value";
+            var args = new IidSpikeDetector.Arguments()
+            {
+                Source = inputColumnName,
+                Name = outputColumnName,
+                Confidence = 95,                // The confidence for spike detection in the range [0, 100]
+                PvalueHistoryLength = size / 4  // The size of the sliding window for computing the p-value
+            };
+
+            // The transformed model.
+            ITransformer model = new IidSpikeEstimator(ml, args).Fit(dataView);
+
+            // Create a time series prediction engine from the model.
+            var engine = model.CreateTimeSeriesPredictionFunction<IidSpikeData, IidSpikePrediction>(ml);
+            for (int index = 0; index < 5; index++)
+            {
+                // Anomaly spike detection.
+                var prediction = engine.Predict(new IidSpikeData(5));
+                Console.WriteLine("{0}\t{1}\t{2:0.00}\t{3:0.00}", 5, prediction.Prediction[0],
+                    prediction.Prediction[1], prediction.Prediction[2]);
```

Comment:
No offense, but this is awful.
How I'm suppose to know size of array I need to pass?
How I'm suppose to know what is meaning behind index 0, index 1, index 2, index 3?

Why this not wrap in a proper class with properties?

Reply:
I think Zeeshan made a note of this in the PR for prediction engine, and mentioned that it would require a bit of extra refactoring so he would make it in a separate PR.  I'll let him elaborate.

---
In reply to: [237329405](https://github.com/dotnet/machinelearning/pull/1762#discussion_r237329405) [](ancestors = 237329405)

Reply:
Do we still want to do the future work to wrap this output vector column in a class with named properties, or even turning it into three scalar output columns?

---
In reply to: [237625679](https://github.com/dotnet/machinelearning/pull/1762#discussion_r237625679) [](ancestors = 237625679,237595774,237329405)